x-pack/plugins/apm/public/utils/formatters.ts

```diff
@@ -23,45 +31,57 @@ interface FormatterOptions {
 
 export function asSeconds(
   value: FormatterValue,
-  { withUnit = true, defaultValue = 'N/A' }: FormatterOptions = {}
+  { withUnit = true, defaultValue = notAvailableLabel }: FormatterOptions = {}
 ) {
   if (value == null) {
     return defaultValue;
   }
+  const secondsLabel =
+    SPACE +
+    i18n.translate('xpack.apm.formatters.secondsTimeUnitLabel', {
+      defaultMessage: 's'
+    });
   const formatted = asDecimal(value / 1000000);
-  return `${formatted}${withUnit ? ' s' : ''}`;
+  return `${formatted}${withUnit ? secondsLabel : ''}`;
 }
 
 export function asMillis(
   value: FormatterValue,
-  { withUnit = true, defaultValue = 'N/A' }: FormatterOptions = {}
+  { withUnit = true, defaultValue = notAvailableLabel }: FormatterOptions = {}
 ) {
   if (value == null) {
     return defaultValue;
   }
 
+  const millisLabel =
+    SPACE +
+    i18n.translate('xpack.apm.formatters.millisTimeUnitLabel', {
+      defaultMessage: 'ms'
+    });
   const formatted = asInteger(value / 1000);
-  return `${formatted}${withUnit ? ' ms' : ''}`;
+  return `${formatted}${withUnit ? millisLabel : ''}`;
 }
 
 export function asMicros(
   value: FormatterValue,
-  { withUnit = true, defaultValue = 'N/A' }: FormatterOptions = {}
+  { withUnit = true, defaultValue = notAvailableLabel }: FormatterOptions = {}
 ) {
   if (value == null) {
     return defaultValue;
   }
 
+  const milcrosLabel =
+    SPACE +
+    i18n.translate('xpack.apm.formatters.microsTimeUnitLabel', {
+      defaultMessage: 'μs'
+    });
   const formatted = asInteger(value);
-  return `${formatted}${withUnit ? ' μs' : ''}`;
+  return `${formatted}${withUnit ? milcrosLabel : ''}`;
 }
 
 type TimeFormatter = (
   max: number
-) => (
-  value: FormatterValue,
-  { withUnit, defaultValue }: FormatterOptions
-) => string;
+) => (value: FormatterValue, options: FormatterOptions) => string;
```

Comment:
`i18n` tool was broken with `{ withUnit, defaultValue }: FormatterOptions` because of babel parser, with error `Name in a signature must be an Identifier.`. So I had to fix it. Destructuring is not necessary here because the property is already described in `FormatterOptions` type.

Reply:
Yes, it's better this way 👍